salt/client/__init__.py

```diff
@@ -1761,7 +1730,8 @@ def pub(self,
             if listen and not self.event.connect_pub(timeout=timeout):
                 raise SaltReqTimeoutError()
             payload = channel.send(payload_kwargs, timeout=timeout)
-        except SaltReqTimeoutError:
+        except SaltReqTimeoutError as err:
+            log.error(err)
```

Comment:
@moio this is intended. That error log entry proved itself very useful while trying to understand why it is timing out. I think it would be a good idea to have it in the logs and not hide it with the general error message that follows.

Reply:
:+1: